### PR TITLE
Add `--pkg-config-path` to `LocationOptions`

### LINKED ISSUE #5815: Time to graduate PKG_CONFIG_PATH to a command-line option?
### Description

The `PKG_CONFIG_PATH` environment variable is used to [specify alternate paths to search for pkg-config `.pc` files](https://github.com/apple/swift-package-manager/blob/17b3ad89e78cc322e96adcbfc0478a86b92d401a/Sources/PackageLoading/PkgConfig.swift#L34), which is useful when building packages against alternate sysroots, eg for cross-compilation. Strangely, SPM does not check the alternate sysroot specified by the `--sdk` flag, which is what I pass to this environment variable again anyway.

This is now [relevant to SPM itself, since it recently began looking for SQLite3 this way in trunk](https://github.com/apple/swift-package-manager/commit/239f3178b8f7f2463a7754d2a802535a9832d320#diff-f913940c58e8744a2af1c68b909bb6383e49007e6c5a12fb03104a9006ae677eR173), so I had to modify my Android CI to start passing in the path to a cross-compiled SQLite3 for Android when cross-compiling SPM, buttaface/swift-android-sdk@f36e045.

### Expected behavior

_No response_

### Actual behavior

_No response_

### Steps to reproduce

_No response_

### Swift Package Manager version/commit hash

_No response_

### Swift & OS version (output of `swift --version && uname -a`)

_No response_

### PR DESCRIPTION
**Pull request by MaxDesiatov**
Resolves #5815.

### Motivation:

Having a specific `--pkg-config-path` option for this means we have something that's less fragile and easier to report errors on than `PKG_CONFIG_PATH` environment variable.

### Modifications:

Added `pkgConfigDirectory` to `LocationOptions`, which is propagated to `BuildPlan` as a function or initializer argument.

### Result:

`--pkg-config-path` is available as a command-line option.

### CODE REVIEW

File: Sources/XCBuildSupport/PIFBuilder.swift [hunk 1]
```diff
@@ -24,23 +24,15 @@ import SPMBuildCore
...
-    /// Creates a `PIFBuilderParameters` instance.
-    /// - Parameters:
-    ///   - enableTestability: Whether or not build for testability is enabled.
-    ///   - shouldCreateDylibForDynamicProducts: Whether to create dylibs for dynamic library products.
-    public init(enableTestability: Bool, shouldCreateDylibForDynamicProducts: Bool, toolchainLibDir: AbsolutePath) {
-        self.enableTestability = enableTestability
-        self.shouldCreateDylibForDynamicProducts = shouldCreateDylibForDynamicProducts
-        self.toolchainLibDir = toolchainLibDir
-    }
+    let pkgConfigDirectory: AbsolutePath?
```
Comment (tomerd) [inline Sources/XCBuildSupport/PIFBuilder.swift:35]: nice cleanup

**Comment by MaxDesiatov** [comment]
> I believe `PKG_CONFIG_PATH` is a list of paths, so if we want to provide an alternative to it, we should probably also take a list here.

Would it make sense to make them comma-separated? I wonder if just changing the type on the option declaration to `[AbsolutePath]` would work with some built-in support from ArgumentParser?

> If I read this correctly, it is also not being passed to XCBuild support, yet, we should do that.

Is there anything to adjust other than this place in `PIFBuilder.swift`? https://github.com/apple/swift-package-manager/pull/5949/files#diff-4558ab2f4d59c923d54773a355350f9362890d5b982aaf79ae5854f8e93eb09fR703

**Comment by neonichu** [comment]
> Would it make sense to make them comma-separated? I wonder if just changing the type on the option declaration to [AbsolutePath] would work with some built-in support from ArgumentParser?

Not sure, but we should do whatever is the most canonical way to do this. Maybe @natecook1000 has hints

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

**Comment by MaxDesiatov** [comment]
> > Would it make sense to make them comma-separated? I wonder if just changing the type on the option declaration to [AbsolutePath] would work with some built-in support from ArgumentParser?
> 
> Not sure, but we should do whatever is the most canonical way to do this.

@neonichu turns out, specifying it as array allows passing the option multiple times. I've also added tests and updated `CHANGELOG.md`, this is now ready for review.

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Comment by buttaface** [comment]
I was hoping we'd get this in before the 5.8 branch tomorrow, still possible?

**Comment by MaxDesiatov** [comment]
I'm not sure, it depends on whether anyone is available for a review.

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,13 @@ Note: This is in reverse chronological order, so newer entries are added to the
 Swift 5.8
 -----------
 
+* [#5949]
```
Comment (tomerd) [inline CHANGELOG.md:6]: I think we should sort these

Comment (MaxDesiatov) [reply CHANGELOG.md:6]: It's already in reverse chronological order, or would you like to see some other sorting?

Comment (tomerd) [reply CHANGELOG.md:6]: I think the rest of the document has is non-reverse chronological order?

**Comment by buttaface** [comment]
I can confirm that [this flag works when cross-compiling SPM for Android with the latest 5.8 Dec. 20 snapshot](https://github.com/buttaface/swift-android-sdk/actions/runs/3756387249/jobs/6382415284#step:6:20), thanks.

**Pull request merged**